### DISCUSSION TITLE
Next image images are not lazy loading after client side transition

### DISCUSSION THREAD

**Discussion by abineetds** [vercel/next.js#56613] [Help]
### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/distracted-hill-ghgmn7

### To Reproduce

1. Open up the code sandbox
2. Right click to open the dev tools in your browser > Go to the network tab > Filter to img requests
3. Click on "Go to next"
4. Click on "Go to home"

### Current vs. Expected behavior

Observed:
All 50 images are loaded when coming back to the home page after the client side transition

Expected:
Page respects lazy loading

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000
Binaries:
  Node: 18.10.0
  npm: 8.19.2
  Yarn: 1.22.18
  pnpm: N/A
Relevant Packages:
  next: 13.5.4
  eslint-config-next: 13.5.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Image optimization (next/image, next/legacy/image)

### Additional context

Hello,

We recently upgraded to Next 13 and started using the new image component. 
We're having trouble with client side routing on pages where we have a lot of images, because all of the images are getting loaded and then getting cached to disk, which in some scenarios is crashing the browser.

https://shop.getbezel.com/ if you want to see it happen in production, but I also whipped up a code sandbox sample.

We have close to 800 pages on our homepage and lean heavily on lazy loading!

Thanks in advance for the help!

**Comment by balazsorban44** [comment]
Hi, image optimization is not enabled in `next dev` mode, but _it is_ in production. So this is at least expected there. In production though, images will be lazily loaded based on how far off they are from the visible viewport. This seems to behave correctly/as expected in the provided CodeSandbox, when using `next build && next start.

However checking https://shop.getbezel.com/, I can see that images are loaded pretty eagerly, although it seem to be unrelated to Next.js. I can see a lot of requests initiated by a `loadImage.js` file which points to Swiper.

Based solely on the reproduction, the behavior seems to be correct. It's not clear how you are using Swiper, so I cannot investigate further. Feel free top open a new issue with a minimal(!) reproduction using Swiper if you still think it's a Next.js bug

**Reply by abineetds** [reply]
Hi @balazsorban44,

Thanks for the reply here, I think there's a chance something is going on with the compatibility with Swiper that is causing this, but I still don't think the images are loading properly in production. Attached is a video of the code sandbox running with `next build && next start` and reproducing the steps mentioned, I still see the 50 images being loaded on client side transition even though the lazy loading works beautifully on first page load. Why does the behavior change when the page first loads vs subsequent loads client-side?


https://github.com/vercel/next.js/assets/10578131/411c5191-4cae-4588-b7ab-f29b20c3b681

**Comment by techUdayMungalpara** [comment]
if we fetch data from API on the client side, we still need to pay https://vercel.com/docs/accounts/plans for image optimization I want to build an image gallery type application
